code.js

```diff
@@ -1,3 +1,69 @@
-function dijkstra(graph, sourceNode) {
-    return [];
-}
+function dijkstra(graph, sourceNode) {
+
+  if (graph.length == 0) {
+      return null;
+  }
+  unvisited = [];
+  dist = {};
+  for (const vertices in graph) {
+      unvisited.push(vertices);
+      //console.log(unvisited);
+      dist[vertices] = Infinity;
+  }
+  dist[sourceNode] = 0;
+  //console.log(dist);
+  while (unvisited.length > 0) {
+      //console.log(unvisited);
+      //console.log(dist);
+      const vertex = unvisited[0];

```

Comment:
Don't you have to ensure that this is the node with the lowest distance estimate?